panc/src/main/scripts/panlint/panlint.py

```diff
@@ -26,6 +26,13 @@
 from colorama import Fore, Style, init as colorama_init
 
 
+class Line(object):
+    def __init__(self, filename, number, text):
```

Comment:
The huge number of places where I see `panlint.Line('',` being called below suggests filename ought to be an optional argument, not the first argument? Or is there a reason for this ordering?

Reply:
In normal use (not unit tests) the filename would always be provided, as it is required for reporting problems, I don't see the value in modifying the constructor just for the benefit of the unit tests. But yes I suppose it could be an optional argument, or ideally further re-factoring could make lines properties of file objects — I'll open an issue.

Reply:
Anyone else want to comment on this / review the code before we merge it?